plugin/telegram.py

```diff
@@ -0,0 +1,113 @@
+@author:      Bastian Schroll
+@description: Template Plugin File
+"""
+import logging
+
+from telegram.error import (TelegramError, Unauthorized, BadRequest, TimedOut, ChatMigrated, NetworkError)
+from plugin.pluginBase import PluginBase
+import telegram
+import re
+import geocoder
+
+# ###################### #
+# Custom plugin includes #
+
+# ###################### #
+
+logging.debug("- %s loaded", __name__)
+
+
+class BoswatchPlugin(PluginBase):
+    """!Description of the Plugin"""
+
+    def __init__(self, config):
+        """!Do not change anything here!"""
+        super().__init__(__name__, config)  # you can access the config class on 'self.config'
+
+    def onLoad(self):
+        """!Called by import of the plugin"""
+        pass
+
+    def setup(self):
+        """!Called before alarm"""
+        self.bot = telegram.Bot(token=self.config.get("botToken", default=""))
+        pass
+
+    def fms(self, bwPacket):
+        """!Called on FMS alarm
+
+        @param bwPacket: bwPacket instance"""
+        logging.warning("Telegram Plugin does not work for FMS")
+        pass
+
+    def pocsag(self, bwPacket):
+        """!Called on POCSAG alarm
+
+        @param bwPacket: bwPacket instance"""
+
+        try:
+            # Send Message via Telegram
+            msg = bwPacket.get("ric") + " (" + bwPacket.get("subric") + ")\n" + bwPacket.get("message")
+            self.bot.send_message(chat_id=self.config.get("chatId", default=""), text=msg)
+
+            # Send Location via Telegram if Geocoding is enabled and Provider and Key are set
+            if self.config.get("geocoding", default=False):
+                address = re.search(self.config.get("geoRegex"), bwPacket.get("message"))[1]
+                provider = self.config.get("geoApiProvider")
+
+                if "mapbox" == provider:
+                    g = geocoder.mapbox(address, key=self.config.get("geoApiToken"))
+                elif "google" == provider:
+                    g = geocoder.google(address, key=self.config.get("geoApiToken"))
+                else:
+                    return
+
+                (lat, lng) = g.latlng
+                self.bot.sendLocation(chat_id=self.config.get("chatId", default=""), latitude=lat, longitude=lng)
+        except (IndexError, TypeError, ValueError):
+            logging.warning("Address was not found in current Message, skipping Location")
+        except Unauthorized:
+            logging.error("Error while sending Telegram Message, please Check your api-key")
+        except (TimedOut, NetworkError):
+            logging.error("Error while sending Telegram Message, please Check your connectivity")
+        except (BadRequest, TelegramError):
+            logging.error("Error while sending Telegram Message")
+        except Exception as e:
+            logging.error("Unknown Error while sending Telegram Message: " + str(type(e).__name__) + ": " + str(e))
+
+        pass
+
+    def zvei(self, bwPacket):
```

Comment:
Nicht implementierte Typen können komplett entfernt werden, die Basisklasse wirft in diesem Fall sowieso eine entsprechende Log Meldung